recipes-omnect/omnect-device-service/omnect-device-service.inc

```diff
@@ -26,20 +23,33 @@ LIC_FILES_CHKSUM = " \
 "
 
 DEPENDS += "\
-    azure-iot-sdk-c\
+    azure-iot-sdk-c \
+    pkgconfig-native \
 "
+
 RDEPENDS:${PN} += "\
     aziot-identityd\
     ca-certificates\
+    polkit \
+    sudo \
+    ${@bb.utils.contains('MACHINE_FEATURES', 'grub', 'grub-editenv', 'u-boot-fw-utils', d)} \
 "
 inherit omnect-device-service omnect_rust_azure-iot-sdk_deps systemd
 
+CARGO_BUILD_FLAGS .= "${@bb.utils.contains('MACHINE_FEATURES', 'grub', ' --features=bootloader_grub', ' --features=bootloader_uboot', d)}"
+
 do_install:append() {
     install -m 0644 -D ${S}/systemd/omnect-device-service.service ${D}${systemd_system_unitdir}/omnect-device-service.service
     install -m 0644 -D ${S}/systemd/omnect-device-service.timer ${D}${systemd_system_unitdir}/omnect-device-service.timer
     install -m 0644 -D ${S}/polkit/omnect-device-service.rules ${D}${datadir}/polkit-1/rules.d/omnect-device-service.rules
     install -m 0640 -D ${S}/sudo/omnect-device-service ${D}${sysconfdir}/sudoers.d/omnect-device-service
 
+    if ${@bb.utils.contains('MACHINE_FEATURES', 'grub', 'true', 'false', d)}; then
+        install -m 0640 -D ${S}/sudo/omnect-device-service-grub ${D}${sysconfdir}/sudoers.d/omnect-device-service-bootloader-env
+    else
+        install -m 0640 -D ${S}/sudo/omnect-device-service-uboot ${D}${sysconfdir}/sudoers.d/omnect-device-service-bootloader-env
+    fi
+
```

Comment:
afaics the source files for installation aren't part of the PR, where can they be found?

Reply:
`${S}` here means its part of source downloaded ... thus the omnect-device-service repo

Reply:
ah, right, should have looked more thoroughly, thx.